src/app/devices/event-channel/event-channel.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, OnDestroy, ViewChild } from '@angular/core'
+import {
+  IMqttMessage,
+  MqttService
+} from 'ngx-mqtt'
+import { Subscription } from 'rxjs'
+import { EventChannel } from 'src/models/models'
+import { MatTableDataSource } from '@angular/material/table'
+import { MatPaginator } from '@angular/material/paginator'
+import { MatSort } from '@angular/material/sort'
+
+@Component({
+  selector: 'app-event-channel',
+  templateUrl: './event-channel.component.html',
+  styleUrls: ['./event-channel.component.scss']
+})
+
+export class EventChannelComponent implements OnDestroy {
+  public subscription!: Subscription
+  public message: string = ''
+  public displayedColumns: string[] = ['guid', 'message', 'type', 'methods', 'timestamp']
+  storedData = localStorage.getItem('dataSource')
+  private readonly cacheData = this.storedData ? JSON.parse(this.storedData) : []
+  public eventChannelLogs: EventChannel = { data: this.cacheData }
+  public dataSource = new MatTableDataSource(this.eventChannelLogs.data)
+
+  @ViewChild(MatPaginator, { static: false }) paginator!: MatPaginator
+  @ViewChild(MatSort, { static: false }) sort!: MatSort
+
+  constructor (private readonly _mqttService: MqttService) {
+    this.subscribeMqtt('mps/#')
+    this.subscribeMqtt('rps/#')
+  }
+
+  subscribeMqtt (event: string): void {
+    this.subscription = this._mqttService.observe(event).subscribe((message: IMqttMessage) => {
+      this.message = message.payload.toString()
+      this.eventChannelLogs.data.splice(0, 0, JSON.parse(this.message))
+      this.eventChannelLogs.data = this.eventChannelLogs.data.slice(0, 100)
```

Comment:
shouldn't this check the length of the data first before slicing only perform this operation if we need to?